apps/ui/modules/util.ts

```diff
@@ -290,39 +290,88 @@ export function getPostLoginUrl() {
   return `${origin}${pathname}?${params.toString()}`;
 }
 
-// TODO move to common and combine with admin (requires i18n changes: replace messages.ts with json)
-export function formatTimeRange(
-  t: TFunction,
-  beginDate: Date,
-  endDate: Date
-): string {
-  const beginTime = t("common:timeWithPrefixInForm", { date: beginDate });
-  const endTime = t("common:timeInForm", { date: endDate });
-  return `${beginTime}–${endTime}`;
+// date format should always be in finnish, but the weekday and time separator should be localized
+const dateFormatParams = {
+  date: {
+    year: "numeric",
+    month: "numeric",
+    day: "numeric",
+    locale: "fi",
+  },
+};
+
+function formatTime(t: TFunction, date: Date): string {
+  return t("common:dateWithWeekday", {
+    date,
+    formatParams: {
+      date: {
+        hour: "numeric",
+        minute: "numeric",
+        hour12: false,
+      },
+    },
+  });
 }
+
+function dayTimeSeparator(t: TFunction): string {
+  return t("common:dayTimeSeparator");
+}
+
 export function formatDateTimeRange(
   t: TFunction,
   begin: Date,
   end: Date
 ): string {
-  const beginDate = t("common:dateWithWeekday", { date: begin });
-  const beginTime = t("common:timeWithPrefixInForm", { date: begin });
-  const endDate = t("common:dateWithWeekday", { date: end });
-  const endTime = t("common:timeInForm", { date: end });
-
-  return trim(
-    `${beginDate} ${beginTime}–${
-      endDate !== beginDate ? endDate : ""
-    }${endTime}`,
-    " – "
-  );
+  // TODO change the key names
+  const beginDate = t("common:dateWithWeekday", {
+    date: begin,
+    formatParams: dateFormatParams,
+  });
+
+  const day = formatDay(t, begin);
+
+  const showEndDate = !isSameDay(begin, end);
+  // TODO this has incorrect formatting when the dates differ (fi format for time separator)
+  const endTime = t("common:dateWithWeekday", {
+    date: end,
+    formatParams: {
+      date: {
+        ...(showEndDate
+          ? dateFormatParams.date
+          : {
+              hour: "numeric",
+              minute: "numeric",
+              hour12: false,
+            }),
```

Comment:
This has a mistake: need to always include the time. Could also fix the separator error (and remove TODO).